Signal/src/ViewControllers/ContactsViewHelper.m

```diff
@@ -199,6 +207,76 @@ - (BOOL)doesSignalAccount:(SignalAccount *)signalAccount matchSearchTerms:(NSArr
         }]];
 }
 
+- (BOOL)doesContact:(Contact *)contact matchSearchTerm:(NSString *)searchTerm
+{
+    OWSAssert(contact);
+    OWSAssert(searchTerm.length > 0);
+
+    if ([contact.fullName.lowercaseString containsString:searchTerm.lowercaseString]) {
+        return YES;
+    }
+
+    NSString *asPhoneNumber = [PhoneNumber removeFormattingCharacters:searchTerm];
+    if (asPhoneNumber.length > 0) {
+        for (PhoneNumber *phoneNumber in contact.parsedPhoneNumbers) {
+            if ([phoneNumber.toE164 containsString:asPhoneNumber]) {
+                return YES;
+            }
+        }
+    }
+
+    return NO;
+}
+
+- (BOOL)doesContact:(Contact *)contact matchSearchTerms:(NSArray<NSString *> *)searchTerms
+{
+    OWSAssert(contact);
+    OWSAssert(searchTerms.count > 0);
+
+    for (NSString *searchTerm in searchTerms) {
+        if (![self doesContact:contact matchSearchTerm:searchTerm]) {
+            return NO;
+        }
+    }
+
+    return YES;
+}
+
+- (NSArray<Contact *> *)nonSignalContactsMatchingSearchString:(NSString *)searchText
+{
+    NSArray<NSString *> *searchTerms = [self searchTermsForSearchString:searchText];
+
+    if (searchTerms.count < 1) {
+        return [NSArray new];
+    }
+
+    return [self.nonSignalContacts filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(Contact *contact,
+                                                                   NSDictionary<NSString *, id> *_Nullable bindings) {
+        return [self doesContact:contact matchSearchTerms:searchTerms];
+    }]];
+}
+
+- (nullable NSArray<Contact *> *)nonSignalContacts
```

Comment:
Well, this value is cached and we clear the cache when contacts change by setting the cached value to nil.